### PR TITLE
feat: introduce ServiceNow integration

### PR DESCRIPTION
**Pull request by vkrizan**
* merges REAMDE files and updates it for new integration
* introduces ServiceNow integration (handler) 
* introduces `BasicAuthenticationProcessor`
  * allows dynamic authentication with a single Camel https component

Important notes:
* https only
* no events ungrouping, events are sent as is
* username for basic authentication is currently constant: `rh_insights_integration`

EVNT-600

### CODE REVIEW

**Review by jpramos123 — APPROVED** [review]
Thanks for this PR!

Tested it locally, and this is working just fine.

Created [EVNT-642](https://issues.redhat.com/browse/EVNT-642) so we can adapt `application.properties` to switch between the integrations

File: README.md [hunk 1]
```diff
@@ -3,9 +3,135 @@
...
+### Setup
+
+Set HTTP Event Collector (HEC) on your (local) Splunk and copy the token.
+Ensure that the HEC has SSL disabled and that all tokens are enabled.
+
+### Container Image Build
+
+```
+$ cd splunk-quarkus
+$ podman build -f Dockerfile.jvm -t quay.io/cloudservices/eventing-integrations-splunk ..
```
Comment (petrblaho) [inline README.md:27]: Are we going to change it in the future to not have `splunk` in the name?

File: README.md [hunk 2]
```diff
@@ -3,9 +3,135 @@
...
+```
+
+ServiceNow event
+```
+{"specversion":"1.0","type":"com.redhat.console.notification.toCamel.servicenow","source":"notifications","id":"9dc9a4b1-8868-4afc-a69d-e8723b20452c","time":"2022-02-02T14:09:55.532551Z","rh-account":"12345","data":"{\"notif-metadata\":{\"url\":\"http://localhost:8088\",\"X-Insight-Token\":\"TOKEN\",\"extras\":\"{}\"},\"account_id\":\"12345\",\"org_id\":\"67890\",\"application\":\"advisor\",\"bundle\":\"rhel\",\"context\":{},\"event_type\":\"new-recommendation\",\"timestamp\":\"2022-02-02T14:09:55.344612\",\"events\":[{\"some\":\"eventdata\"}]}"}
+```
+(don't forget to replace `TOKEN` with your token or password)
```
Comment (petrblaho) [inline README.md:95]: :+1:

File: splunk-quarkus/src/main/java/com/redhat/console/integrations/BasicAuthenticationProcessor.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        JsonObject metadata = in.getHeader("metadata", JsonObject.class);
+        String password = metadata.getString("X-Insight-Token");
+        if (password == null || password.equals("")) {
+            return;
+        }
+
+        Credentials credentials = new UsernamePasswordCredentials(this.username, password);
+
+        BasicScheme basicScheme = new BasicScheme();
+        Header header = basicScheme.authenticate(credentials, StandardCharsets.US_ASCII.name(), false /* proxy **/);
```
Comment (petrblaho) [inline splunk-quarkus/src/main/java/com/redhat/console/integrations/BasicAuthenticationProcessor.java:43]: This one particular method is deprecated - https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/auth/BasicScheme.java#L194

**Review by petrblaho — CHANGES_REQUESTED** [review]
Only the deprecated one is issue now...

File: README.md (hunk 1 above)
Comment (vkrizan) [reply README.md:27]: Yes!

File: splunk-quarkus/src/main/java/com/redhat/console/integrations/BasicAuthenticationProcessor.java (hunk 3 above)
Comment (vkrizan) [reply splunk-quarkus/src/main/java/com/redhat/console/integrations/BasicAuthenticationProcessor.java]: Changed to use the non-deprecated method. Created `BasicHttpRequest` and `BasicHttpContext` for it.

**Review by petrblaho — APPROVED** [review]
LGTM!

**Pull request merged**